packages/tailwindcss/src/compat/theme-variants.ts

```diff
@@ -0,0 +1,63 @@
+import { rule } from '../ast'
+import type { DesignSystem } from '../design-system'
+import type { ResolvedConfig } from './config/types'
+
+export function registerThemeVariantOverrides(config: ResolvedConfig, designSystem: DesignSystem) {
+  let ariaVariants = config.theme.aria || {}
+  let supportsVariants = config.theme.supports || {}
+  let dataVariants = config.theme.data || {}
+
+  if (Object.keys(ariaVariants).length > 0) {
+    let coreAria = designSystem.variants.get('aria')
+    let coreApplyFn = coreAria?.applyFn
+    let coreCompounds = coreAria?.compounds
+    designSystem.variants.functional(
+      'aria',
+      (ruleNode, variant) => {
+        let value = variant.value
+        if (value && value.kind === 'named' && value.value in ariaVariants) {
+          ruleNode.nodes = [rule(`&[aria-${ariaVariants[value.value]}]`, ruleNode.nodes)]
+          return
+        }
+        return coreApplyFn?.(ruleNode, variant)
```

Comment:
I think this could be written like so:
```ts
let value = variant.value

if (value && value.kind === 'named' && value.value in ariaVariants) {
  return coreApplyFn?.(ruleNode, {
    ...variant,
    value: {
      kind: 'arbitrary',
      value: ariaVariants[value.value] as string,
    },
  })
}

return coreApplyFn?.(ruleNode, variant)
```

which would mean that the core variant is still the thing that decides how it's formatted which I think works the same way it did in v3.

Also means that the logic between all of these is identical _except_ for the list of values it checks.